repose-aggregator/artifacts/valve/src/main/scala/org/openrepose/valve/ReposeJettyServer.scala

```diff
@@ -126,37 +126,32 @@ class ReposeJettyServer(val nodeContext: AbstractApplicationContext,
 
         //Handle the Protocols and Ciphers
         //varargs are annoying, so lets deal with this using the scala methods
+        // From https://www.eclipse.org/jetty/documentation/9.4.x/configuring-ssl.html
+        // "When working with Includes / Excludes, it is important to know that Excludes will always win."
         import scala.collection.JavaConversions._
-        Option(ssl.getExcludedProtocols).foreach(xp => cf.addExcludeProtocols(xp.getProtocol.toList: _*))
-        Option(ssl.getIncludedProtocols).foreach { ip =>
-          if (ip.getProtocol.nonEmpty) {
-            cf.setIncludeProtocols(ip.getProtocol.toList: _*)
-          }
-
-        }
-        Option(ssl.getExcludedCiphers).foreach { xc =>
-          cf.addExcludeCipherSuites(xc.getCipher.toList: _*)
-        }
-
-        Option(ssl.getIncludedCiphers).foreach { ic =>
-          if (ic.getCipher.nonEmpty) {
-            cf.setIncludeCipherSuites(ic.getCipher.toList: _*)
-          }
-        }
+        // Clear Jetty defaults that were added in later v9.2.x releases.
+        cf.setIncludeProtocols()
+        cf.setExcludeProtocols()
+        cf.setIncludeCipherSuites()
+        cf.setExcludeCipherSuites()
+        Option(ssl.getIncludedProtocols) foreach { ip => cf.setIncludeProtocols(ip.getProtocol.toList: _*) }
+        Option(ssl.getExcludedProtocols) foreach { ep => cf.setExcludeProtocols(ep.getProtocol.toList: _*) }
+        Option(ssl.getIncludedCiphers) foreach { ic => cf.setIncludeCipherSuites(ic.getCipher.toList: _*) }
+        Option(ssl.getExcludedCiphers).foreach { ec => cf.setExcludeCipherSuites(ec.getCipher.toList: _*) }
```

Comment:
Woo! I like it.

So two things to track from this:

1. We should write a story to allow empty include/exclude sequences in the container configuration.
1. For Repose 9, we should decide if we want to change the default include/exclude protocols/ciphers. That may mean defaulting to the same defaults as Jetty, defaulting to whatever is supported by the environment, or defaulting to our own set of defaults.

Reply:
Concur.

Reply:
Wait, why is there a rando `.foreach` on the last set here?

Reply:
Oh, also, I think you could use the following if it makes things clearer:
```
Option(ssl.getExcludedCiphers).map(_.getCipher.toList).foreach(cf.setExcludeCipherSuites)
```